### ISSUE TITLE
Search amongst multiple models

### ISSUE THREAD

**Issue by sicaboy** [babenkoivan/elastic-scout-driver-plus#6]
Does the package plan to have a function similar to matchish's **[Search amongst multiple models](https://github.com/matchish/laravel-scout-elasticsearch#search-amongst-multiple-models)** ?

**Comment by davorminchorov** [comment]
I am looking for something like this as well, to be able to implement global search in one query rather than merging the queries.

**Comment by babenkoivan** [comment]
Hey @sicaboy and @davorminchorov, it's a good idea to add this feature in the package. I'll put it in my `todo` list and try to investigate how and when it can be delivered.

**Comment by marcoboers** [comment]
I am currently working on this. I will probably be able to create a pull request by tomorrow.

**Comment by davorminchorov** [comment]
@marcoboers If you have an index `products` and have 2 document types `shoes` and `clothes`, I am assuming it will search by multiple document types according to your title. Is this correct? 

Also, if you have a simpler structure like indices of `users` and `groups`, it won't be able to cover search by both of them, right?

**Comment by marcoboers** [comment]
@davorminchorov that's correct. It allows you to search multiple document types in a single index. This because scoring differs (most times slightly) per index, and in this way it is possible to sort multiple document types based on their score.

While it searches a single index, it still allows different fields per document type and therefore also searching document type specific fields.

**Comment by babenkoivan** [comment]
Hi guys, just a short update on the topic. Unfortunately, I've decided not to merge https://github.com/babenkoivan/elastic-scout-driver-plus/pull/8, if you would like to know the reason, check the PR comments. I've started working on a concept for the native multiple index search support and I hope, that I can pull everything together within the following 2 weeks.

**Comment by babenkoivan** [comment]
Hi guys, I've just released [v1.5](https://github.com/babenkoivan/elastic-scout-driver-plus/releases/tag/v1.5.0) with multiple index search support. There is a new method in your disposal - `join`, which you can use to add another searchable model (or several separated by comma) to the search request. Here is a usage example:

```
$found = Author::boolSearch()
            ->join(Book::class)
            ->should('match', ['name' => 'John'])
            ->should('match', ['title' => 'The Best Book'])
            ->minimumShouldMatch(1)
            ->execute();
```

In this example, the `title` field belongs to the `Book` model and the `name` field belongs to `Author`. Both models will be returned in a single collection:

```
$models = $found->models();
```

I'm going to rework the documentation (README) in the near future and add more usage examples 🙂
